File: twintilt/app/src/main/java/com/koodipuukko/twintilt/TTGLSurfaceView.java
package com.koodipuukko.twintilt;

import android.content.Context;
import android.opengl.GLSurfaceView;
import android.view.MotionEvent;

public class TTGLSurfaceView
  extends GLSurfaceView
  {
    private final GLRenderer mRenderer;

    public TTGLSurfaceView(Context context, GLRenderer renderer)
      {
        super(context);
        // Set the Renderer for drawing on the GLSurfaceView
        mRenderer = renderer;
        setRenderer(mRenderer);
        // Render the view only when there is a change in the drawing data
        setRenderMode(GLSurfaceView.RENDERMODE_CONTINUOUSLY);
      }
  }

File: app/src/main/java/com/koodipuukko/twintilt/TTGLSurfaceView.kt
package com.koodipuukko.twintilt

import android.content.Context
import android.opengl.GLSurfaceView
import android.util.AttributeSet

class TTGLSurfaceView(context: Context?, private val mRenderer: GLRenderer) :
	GLSurfaceView(context) {
	init {
		setRenderer(mRenderer)
		renderMode = RENDERMODE_CONTINUOUSLY
	}

	constructor(context: Context?) : this(context, GLRenderer(context!!)) {
	}

	constructor(context: Context?, attrs: AttributeSet?) : this(context, GLRenderer(context!!)) {
	}

	constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : this(context, GLRenderer(context!!)) {
	}
}
